README.md

```diff
@@ -0,0 +1,116 @@
+| Trapezoid Rule                  | Primitive        | [0,1]   | `UniformTrapezoid`  |
+| Becke                           | Radial           | (0,inf) | `Becke`             |
+| Murray-Handy-Laming (MHL)       | Radial           | (0,inf) | `MurrayHandyLaming` |
+| Mura-Knowles (MK)               | Radial           | (0,inf) | `MuraKnowles`       |
+| Treutler-Ahlrichs (TA, M3 + M4) | Radial           | (0,inf) | `TreutlerAhlrichs`  |
+| Ahrens-Beylkin                  | Angular          | $S^2$   | `AhrensBeylkin`     |
+| Delley                          | Angular          | $S^2$   | `Delley`            |
+| Lebedev-Laikov                  | Angular          | $S^2$   | `LebedevLaikov`     |
+| Womersley                       | Angular          | $S^2$   | `Womersley`         |
+
+
+For the generation of spherical quadratures, IntegratorXX additionally supports the following radial pruning schemes:
+| Name      | Description                          | C++ Specifier             |
+|-----------|--------------------------------------|---------------------------|
+| Unpruned  | No not perform pruning               | `PruningScheme::Unpruned` |
+| Robust    | The Psi4 "robust" pruning scheme     | `PruningScheme::Robust`   |
+| Treutler  | The Treutler-Ahlrichs pruning scheme | `PruntinScheme::Treutler` |
+
+### A Note on Angular Quadratures
+
+All of the currently implemented angular quadrature schemes are only compatible with *specific* grid
+orders corresponding to *specific* algebraic orders of spherical harmonics they integrate exactly.
+The construction of the angular grids takes the number of points as argument, and will fail if the
+grid order is incompatible. As these *magic numbers* are different for each of the quadratures,
+we provide a set of look-up functions which can safely produce compatible grid orders:
+
+```
+using angular_type = LebedevLaikov<double>; // FP64 LL grid, similar for other implementations
+using traits = IntegratorXX::quadrature_traits<angular_type>;
+
+auto npts  = traits::npts_by_algebraic_order(order); // Return the grid order associated with a particular algebraic order
+auto order = traits::algebraic_order_by_npts(npts);  // Return the algebratic order associated with a grid order
+auto next_order = traits::next_algebraic_order(order); // Return the next largest (inclusive) algebratic order compatible with `angular_type` 
+```
+
+
+## Example Usage
+
+Many example usages for 1-d quadratures (i.e. primitive and radial) can be found in `test/1d_quadratures.cxx` and `test/spherical_generator.cxx`. Below is a simple invocation example for the generation of an atomic sphere via the runtime generator:
+```
+using namespace IntegratorXX;               // Import namespace
+auto rad_spec = RadialQuad::MuraKnowles;    // MK Radial scheme
+auto ang_spec = AngularQuad::AhrensBeylkin; // AH Angular scheme
+size_t nrad   = 99;
+size_t nang   = 372;
+double rscal  = 2.0;
```

Comment:
My issue with this is that this requires hardcoding the allowed options. I would like to have alternative functionality, where the library takes in a string specification for the rule and returns the radial or angular specification. That way the downstream interfaces don't have to be changed when new rules are added in IntegratorXX. If this is already possible, please provide an example

Reply:
Based on where you placed this comment, I'm unclear what you mean here - do you mean the specification of the grid orders / scaling factors or the usage of `enums` to specify the the radial / angular rules? 

If you want a string-driven "getter" of the radial/angular specifications, that's fine, but the internal implementations  _**must**_ be discrete, as we have to generate the grids using compile time logic. For this purpose, `enum` == `int` (you'll also notice that I manually specify the integer encodings rather than letting them be auto generate - this is looking forward to making the C/C++ interfaces compatible).

I'd be fine with something like the following:
```
RadialQuad radial_scheme_from_string(std::string name);
AngularQuad angular_scheme_from_string(std::string name);
...
auto rad_spec = radial_scheme_from_string("MuraKnowles");
auto ang_spec = angular_scheme_from_string("AhrensBeylkin");
```
Would that resolve the issue you raise?

Reply:
Added as of https://github.com/wavefunction91/IntegratorXX/pull/90/commits/90a855fb3d900c3bd7850a22f39e9b513b09b273

Reply:
Yes. I was just aiming at it being very hard to extend hardcoded implementations. Having the parser logic in the library makes downstream codes future proof, since you can even allow user-side specification of the input string, kind of how functionals are (or should be!) exlusively handled by libxc...